pyro/distributions/mixture.py

```diff
@@ -102,14 +102,18 @@ def expand(self, batch_shape):
 
     def sample(self, sample_shape=torch.Size()):
         mask = self.mask.expand(sample_shape + self.batch_shape) if sample_shape else self.mask
-        result = self.component0.sample(sample_shape)
-        result[mask] = self.component1.sample(sample_shape)[mask]
+        mask = mask.reshape(mask.shape + (1,) * len(self.event_shape))
+        result = torch.where(mask,
+                             self.component1.sample(sample_shape),
+                             self.component0.sample(sample_shape))
         return result
 
     def rsample(self, sample_shape=torch.Size()):
         mask = self.mask.expand(sample_shape + self.batch_shape) if sample_shape else self.mask
```

Comment:
I think this should be
```py
mask = self.mask.reshape(self.mask.shape + (1,) * self.event_dim)
mask = mask.expand(sample_shape + self.shape())
```

Reply:
@fritzo that looks good to me. I will submit the commit.